### PR TITLE
Use babel loader and update typescript

### LINKED ISSUE #268: An in-range update of redux-saga is breaking the build 🚨
## The dependency [redux-saga]() was updated from `1.0.5` to `1.1.0`.

🚨 [View failing branch](https://github.com/crazyfactory/ts-react-boilerplate/compare/master...crazyfactory:greenkeeper%2Fredux-saga-1.1.0).

This version is **covered** by your **current version range** and after updating it in your project **the build failed**.

redux-saga is a direct dependency of this project, and **it is very likely causing it to break**. If other packages depend on yours, this update is probably also breaking those in turn.

<details>
<summary>Status Details</summary>

- ❌ **Travis CI - Branch:** <a href='https://travis-ci.com/crazyfactory/ts-react-boilerplate/builds/127897014'> The build</a> **failed**. 
</details>

---

<details>
<summary>FAQ and help</summary>

There is a collection of [frequently asked questions](https://greenkeeper.io/faq.html). If those don’t help, you can always [ask the humans behind Greenkeeper](https://github.com/greenkeeperio/greenkeeper/issues/new).
</details>

---

Your [Greenkeeper](https://greenkeeper.io) Bot :palm_tree:

### LINKED ISSUE #288: An in-range update of @typescript-eslint/eslint-plugin is breaking the build 🚨
## The devDependency [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint) was updated from `2.3.3` to `2.4.0`.

🚨 [View failing branch](https://github.com/crazyfactory/ts-react-boilerplate/compare/master...crazyfactory:greenkeeper%2F%40typescript-eslint%2Feslint-plugin-2.4.0).

This version is **covered** by your **current version range** and after updating it in your project **the build failed**.

@typescript-eslint/eslint-plugin is a devDependency of this project. It **might not break your production code or affect downstream projects**, but probably breaks your build or test tools, which may **prevent deploying or publishing**.

<details>
<summary>Status Details</summary>

- ❌ **Travis CI - Branch:** <a href='https://travis-ci.com/crazyfactory/ts-react-boilerplate/builds/131839398'> The build</a> **failed**. 
</details>

---

<details>
<summary>Release Notes for v2.4.0</summary>

<h1><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/compare/v2.3.3...v2.4.0">2.4.0</a> (2019-10-14)</h1>
<h3>Bug Fixes</h3>
<ul>
<li><strong>eslint-plugin:</strong> [promise-function-async] Should not report… (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/issues/1023" data-hovercard-type="pull_request" data-hovercard-url="/typescript-eslint/typescript-eslint/pull/1023/hovercard">#1023</a>) (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/514bed9">514bed9</a>)</li>
<li>support long running "watch" lint sessions (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/issues/973" data-hovercard-type="pull_request" data-hovercard-url="/typescript-eslint/typescript-eslint/pull/973/hovercard">#973</a>) (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/854620e">854620e</a>)</li>
</ul>
<h3>Features</h3>
<ul>
<li><strong>typescript-estree:</strong> support for parsing 3.7 features (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/issues/1045" data-hovercard-type="pull_request" data-hovercard-url="/typescript-eslint/typescript-eslint/pull/1045/hovercard">#1045</a>) (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/623febf">623febf</a>)</li>
</ul>
</details>

<details>
<summary>Commits</summary>
<p>The new version differs by 8 commits.</p>
<ul>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/111ecc668eb8a332d7311dacf196fceec83316cb"><code>111ecc6</code></a> <code>chore: publish v2.4.0</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/623febfca0b5afbbdc7297d65617d80ac4872513"><code>623febf</code></a> <code>feat(typescript-estree): support for parsing 3.7 features (#1045)</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/854620e9232d4d54e6fc89876b4c9ed38f5570f9"><code>854620e</code></a> <code>fix: support long running "watch" lint sessions (#973)</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/fec73b0882dc57001c6b75aa2e837a0557889e70"><code>fec73b0</code></a> <code>docs: clarify which versions of ESLint are supported (#1073)</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/5f92b9f0b21eb45c11029ca9fe18418f73d45a12"><code>5f92b9f</code></a> <code>docs: add TOC to root README (#838)</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/526d336adec7ae173fb259b18b7263859f61539b"><code>526d336</code></a> <code>chore: add test for function types &amp; no-restricted-globals rule (#1055)</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/514bed95fd68d75eecdf3719e52bfac0f3cd8fc2"><code>514bed9</code></a> <code>fix(eslint-plugin): [promise-function-async] Should not report… (#1023)</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/a3f84e196dfe57c380f90f6222a3cdcf25a62bfc"><code>a3f84e1</code></a> <code>chore(eslint-plugin): Add missing rule function types (#1047)</code></li>
</ul>
<p>See the <a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/compare/054df278d6b7064a44b5f78fec453bf9ae6ad281...111ecc668eb8a332d7311dacf196fceec83316cb">full diff</a></p>
</details>

<details>
<summary>FAQ and help</summary>

There is a collection of [frequently asked questions](https://greenkeeper.io/faq.html). If those don’t help, you can always [ask the humans behind Greenkeeper](https://github.com/greenkeeperio/greenkeeper/issues/new).
</details>

---

Your [Greenkeeper](https://greenkeeper.io) Bot :palm_tree:

### LINKED ISSUE #289: An in-range update of @typescript-eslint/parser is breaking the build 🚨
## The devDependency [@typescript-eslint/parser](https://github.com/typescript-eslint/typescript-eslint) was updated from `2.3.3` to `2.4.0`.

🚨 [View failing branch](https://github.com/crazyfactory/ts-react-boilerplate/compare/master...crazyfactory:greenkeeper%2F%40typescript-eslint%2Fparser-2.4.0).

This version is **covered** by your **current version range** and after updating it in your project **the build failed**.

@typescript-eslint/parser is a devDependency of this project. It **might not break your production code or affect downstream projects**, but probably breaks your build or test tools, which may **prevent deploying or publishing**.

<details>
<summary>Status Details</summary>

- ❌ **Travis CI - Branch:** <a href='https://travis-ci.com/crazyfactory/ts-react-boilerplate/builds/131840497'> The build</a> **errored**. 
</details>

---

<details>
<summary>Release Notes for v2.4.0</summary>

<h1><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/compare/v2.3.3...v2.4.0">2.4.0</a> (2019-10-14)</h1>
<h3>Bug Fixes</h3>
<ul>
<li><strong>eslint-plugin:</strong> [promise-function-async] Should not report… (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/issues/1023" data-hovercard-type="pull_request" data-hovercard-url="/typescript-eslint/typescript-eslint/pull/1023/hovercard">#1023</a>) (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/514bed9">514bed9</a>)</li>
<li>support long running "watch" lint sessions (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/issues/973" data-hovercard-type="pull_request" data-hovercard-url="/typescript-eslint/typescript-eslint/pull/973/hovercard">#973</a>) (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/854620e">854620e</a>)</li>
</ul>
<h3>Features</h3>
<ul>
<li><strong>typescript-estree:</strong> support for parsing 3.7 features (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/issues/1045" data-hovercard-type="pull_request" data-hovercard-url="/typescript-eslint/typescript-eslint/pull/1045/hovercard">#1045</a>) (<a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/623febf">623febf</a>)</li>
</ul>
</details>

<details>
<summary>Commits</summary>
<p>The new version differs by 8 commits.</p>
<ul>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/111ecc668eb8a332d7311dacf196fceec83316cb"><code>111ecc6</code></a> <code>chore: publish v2.4.0</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/623febfca0b5afbbdc7297d65617d80ac4872513"><code>623febf</code></a> <code>feat(typescript-estree): support for parsing 3.7 features (#1045)</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/854620e9232d4d54e6fc89876b4c9ed38f5570f9"><code>854620e</code></a> <code>fix: support long running "watch" lint sessions (#973)</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/fec73b0882dc57001c6b75aa2e837a0557889e70"><code>fec73b0</code></a> <code>docs: clarify which versions of ESLint are supported (#1073)</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/5f92b9f0b21eb45c11029ca9fe18418f73d45a12"><code>5f92b9f</code></a> <code>docs: add TOC to root README (#838)</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/526d336adec7ae173fb259b18b7263859f61539b"><code>526d336</code></a> <code>chore: add test for function types &amp; no-restricted-globals rule (#1055)</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/514bed95fd68d75eecdf3719e52bfac0f3cd8fc2"><code>514bed9</code></a> <code>fix(eslint-plugin): [promise-function-async] Should not report… (#1023)</code></li>
<li><a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/commit/a3f84e196dfe57c380f90f6222a3cdcf25a62bfc"><code>a3f84e1</code></a> <code>chore(eslint-plugin): Add missing rule function types (#1047)</code></li>
</ul>
<p>See the <a href="https://urls.greenkeeper.io/typescript-eslint/typescript-eslint/compare/054df278d6b7064a44b5f78fec453bf9ae6ad281...111ecc668eb8a332d7311dacf196fceec83316cb">full diff</a></p>
</details>

<details>
<summary>FAQ and help</summary>

There is a collection of [frequently asked questions](https://greenkeeper.io/faq.html). If those don’t help, you can always [ask the humans behind Greenkeeper](https://github.com/greenkeeperio/greenkeeper/issues/new).
</details>

---

Your [Greenkeeper](https://greenkeeper.io) Bot :palm_tree:

### LINKED ISSUE #290: Migrate from ts-loader to babel-loader
Without type check, it's much faster, and the ecosystem is bigger (a lot of plugins). And the tradeoff is very small. Specifically we can't use
- namespaces - never use anyway
- casting with `<newtype>x` syntax - never use anyway
- enum merging - could be useful in some cases, but rarely used
- legacy-style import `import foo = require(...)` / `export = foo`  - some libraries still export like this, but very small number.

### LINKED ISSUE #294: oc transpilation doesn't work with dynamic property
Just a heads up. This won't work.
```javascript
oc(product).formula.especialnesses["diameter"][diameter].extra_charge(0)
```

Need to change to
```javascript
const extraDiameters = oc(product).formula.especialnesses.diameter(null);
    return extraDiameters !== null && extraDiameters[diameter] !== undefined
      ? extraDiameters[diameter].extra_charge
      : 0;
```

Seems to be a bug of oc typescript transformer. This can't and/or won't be fixed until we migrate to babel loader.

### LINKED ISSUE #299: update to typescript 3.7
for some reason greenkeeper isn't working, I think we should do it manually for this at least, maybe upgrade everything at once and close all PRs,

also typescript 3.7 introduces optional chaining and null coalescing operator, so we'll be removing ts-optchain and this will do the following:

closes #294 
resolve https://github.com/crazyfactory/ts-react-boilerplate/pull/287#issuecomment-548706051 comment for adding ts-optchain as dependency.

### LINKED ISSUE #300: Use cross-fetch instead of isomorphic-fetch
isomorphic-fetch is dead

### PR DESCRIPTION
**Pull request by paibamboo**
Closes #268 
Closes #289 
Closes #288 
Closes #290 
Closes #294 
Closes #299 
Closes #300

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ const tslintRules = {
   ],
   "hex-format": true,
   "import-spacing": true,
-  "interface-sort-keys": true,
+  "interface-sort-keys": false,
```
Comment (paibamboo) [inline .eslintrc.js]: Actually `member-ordering` already has interface sort feature. Using `interface-sort-keys` along with `member-ordering` works great until you want to add methods to an interface.

For example
```typescript
interface ISomething {
   xxxField: string;
   yyyField: string;
   aaaMethod(): string; // interface-sort-keys won't like this.
}
```
and
```typescript
interface ISomething {
   aaaMethod(): string; // member ordering won't like this.
   xxxField: string;
   yyyField: string;
}
```

File: .eslintrc.js [hunk 2]
```diff
@@ -118,16 +135,8 @@ const tslintRules = {
...
-    [
-      {
-        "file": "./src/app/containers/Html.tsx",
-        "method": "render",
-        "comment": "string comes from trusted source - our initial state"
-      }
-    ]
-  ],
+   // todo: Specifying excluded files as an option used to work until updating eslint and it's parsers.
+  "react-no-dangerous-html": true,
```
Comment (paibamboo) [inline .eslintrc.js]: As commented, can't get to work after update eslint and parsers. And not surprisingly the documented version is still [not working](https://github.com/microsoft/tslint-microsoft-contrib/issues/880) and I don't expect it to be fixed. [eslint plugin react no danger rule](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-danger.md) also does not have the option to specify excluded files/locations.

File: src/app/sagas/SettingsSaga.test.ts [hunk 3]
```diff
@@ -50,16 +47,4 @@ describe("SettingsSaga", () => {
       });
     });
   });
-
-  describe("registerListeners", () => {
```
Comment (paibamboo) [inline src/app/sagas/SettingsSaga.test.ts]: Encounter error `Cannot set property fork of #<Object> which has only a getter` so I'm removing this test for now and will get back to it when time permits.

File: src/app/helpers/LanguageHelper.ts [hunk 4]
```diff
@@ -27,17 +27,17 @@ export class LanguageHelper {
     return JSON.parse(fs.readFileSync(LanguageHelper.getLanguageFileLocation(language)).toString());
   }
 
-  public getTranslations(): any {
-    return LanguageHelper.getTranslations(
-      LanguageHelper.isSupported(this.preferredLanguage) ? this.preferredLanguage : "en"
-    );
+  public getPreferredLanguage(): string {
```
Comment (cyberhck) [inline src/app/helpers/LanguageHelper.ts]: this is exactly the same, right? no changes?

**Comment by cyberhck** [comment]
I'm gonna go ahead and merge this, :)

File: src/app/helpers/LanguageHelper.ts (hunk 4 above)
Comment (paibamboo) [reply src/app/helpers/LanguageHelper.ts]: No change

**Pull request merged**